packages/nc-gui/utils/parsers/CSVTemplateAdapter.ts

```diff
@@ -83,8 +83,9 @@ export default class CSVTemplateAdapter {
   detectInitialUidt(v: string) {
     const emailRegex = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/
     const urlRegex = /^(ftp|http|https):\/\/[^ "]+$/
-    const phoneRegex = /\+(9[976]\d|8[987530]\d|6[987]\d|5[90]\d|42\d|3[875]\d|2[98654321]\d|9[8543210]|8[6421]|6[6543210]|5[87654321]|4[987654310]|3[9643210]|2[70]|7|1)\d{1,14}$/
-    if (phoneRegex.test(v)) return UITypes.PhoneNumber
+    const phoneRegex = /^\+(?:[0-9]?){1,3}[-.\s]?(?:\(\d+\)|[0-9]+[-.\s]?)*[0-9]+$/
+    const cleanedV = v.replace(/^'+/, '')
```

Comment:
why we need this line? can you provide some examples?

Reply:
Sure, so when I used only the above phoneRegex, without the words /b/, I was not getting the desired result as phoneNumber from phoneRegex I found out that I was getting the data in v as '+918967453423 and as per regex it's not a valid phoneNumber as it starts with an extra ' in starting so I added this line to exactly remove the extra ' from v and check only the cleanedV in phoneNumber if its a phoneNumber it would be returned if not normal cases would be executed

Reply:
Why `'` specifically?  

Besides, your regex doesn't work with following phone numbers.

```
(877) 359-6695
800-692-7753
0365987456
```

Reply:
> Why `'` specifically?
> 
> Besides, your regex doesn't work with following phone numbers.
> 
> ```
> (877) 359-6695
> 800-692-7753
> 0365987456
> ```

as phone regex is treated as a string, but when I checked the data (v), I found out that an extra ' is there at the start of any phone number, which makes the regex pattern different from that of the phone number, and just by removing that extra ' at the time of checking the cases, it works fine.

Reply:
Can you provide your test csv? Also have you tested the above patterns? I was using https://regex101.com/ to verify your regex.

Reply:
> 
[test.csv](https://github.com/nocodb/nocodb/files/12305887/test.csv)

Sure, yes, I tested it on the website you shared to verify the regex and got a positive response.

Reply:
Failing

```
1-(123) 456-7890
1 (123) 456 7890
+1-222-333-4444
1-(123)-456-7890
```